### PR TITLE
Use "Element Segment" terminology in js-api tests

### PR DESCRIPTION
**Pull request by binji**
This matches the nomenclature of the spec.

Also:

* Remove the `function_table` array from `wasm-module-builder.js` which
  is not used.
* The `appendToTable` function should reference
  `function_table_length` instead of `function_table.length`.

### CODE REVIEW

**Comment by littledan** [comment]
@Ms2ger could you review this PR?

**Comment by binji** [comment]
Rebased.

**Comment by binji** [comment]
I made these changes outside of v8. I can land in v8 first, though.

**Comment by gahaas** [comment]
@Ms2ger is correct, `wasm-module-builder.js` is just a copy of the V8 version where I removed experimental features.

**Comment by binji** [comment]
I landed the v8 change [here](https://chromium-review.googlesource.com/c/v8/v8/+/1273612), and have updated this copy to be the same (minus future features).

File: test/js-api/jsapi.js [hunk 1]
```diff
@@ -1111,7 +1111,6 @@ const complexTableReExportingModuleBinary = (() => {
             46
         ]).index;
 
-    builder.setFunctionTableLength(3);
```
Comment (Ms2ger) [inline test/js-api/jsapi.js]: Why this change?

Comment (binji) [reply test/js-api/jsapi.js]: It's unnecessary because `appendToTable` will set the correct value for the table length. Previously `appendToTable` would use `function_table.length` to determine the offset of the element segment, but `function_table` was never changed, so it would always use 0. Now `appendToTable` uses `table_length_min`, which would be changed by `setTableLength`. This would make a 6-element table with the first 3 elements `null`.
